osu.Game/Screens/Ranking/Statistics/PerformanceBreakdownChart.cs

```diff
@@ -0,0 +1,243 @@
+                                        Text = "Achieved PP",
+                                        Colour = Color4Extensions.FromHex("#66FFCC")
+                                    },
+                                    achievedPerformance = new OsuSpriteText
+                                    {
+                                        Origin = Anchor.CentreRight,
+                                        Anchor = Anchor.CentreRight,
+                                        Font = OsuFont.GetFont(weight: FontWeight.SemiBold, size: 18),
+                                        Colour = Color4Extensions.FromHex("#66FFCC")
+                                    }
+                                },
+                                new Drawable[]
+                                {
+                                    new OsuSpriteText
+                                    {
+                                        Origin = Anchor.CentreLeft,
+                                        Anchor = Anchor.CentreLeft,
+                                        Font = OsuFont.GetFont(weight: FontWeight.Regular, size: 18),
+                                        Text = "Maximum",
+                                        Colour = OsuColour.Gray(0.7f)
+                                    },
+                                    maximumPerformance = new OsuSpriteText
+                                    {
+                                        Origin = Anchor.CentreLeft,
+                                        Anchor = Anchor.CentreLeft,
+                                        Font = OsuFont.GetFont(weight: FontWeight.Regular, size: 18),
+                                        Colour = OsuColour.Gray(0.7f)
+                                    }
+                                }
+                            }
+                        },
+                        chart = new GridContainer
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            Origin = Anchor.TopCentre,
+                            Anchor = Anchor.TopCentre,
+                            ColumnDimensions = new[]
+                            {
+                                new Dimension(GridSizeMode.AutoSize),
+                                new Dimension(),
+                                new Dimension(GridSizeMode.AutoSize)
+                            }
+                        }
+                    }
+                }
+            };
+
+            spinner.Show();
+
+            new PerformanceBreakdownCalculator(playableBeatmap, difficultyCache, performanceCache)
+                .CalculateAsync(score, cancellationTokenSource.Token)
+                .ContinueWith(t => Schedule(() => setPerformanceValue(t.GetResultSafely())));
+        }
+
+        private void setPerformanceValue(PerformanceBreakdown breakdown)
+        {
+            spinner.Hide();
```

Comment:
I haven't managed to get this spinner to ever appear so it's hard to test whether it displays correctly or not. Were you manually adding delays to test this out?

Reply:
Yep. It is working fine. The chart is taller than the spinner, causing a height change when it finishes loading, but I guess it's not noticeable anyway.